services/wallet/service.go

```diff
@@ -493,45 +502,54 @@ func (service *Service) LinkZebPayWallet(ctx context.Context, walletID uuid.UUID
 	return claims.CountryCode, nil
 }
 
-// LinkGeminiWallet links a wallet and transfers funds to newly linked wallet
+const errNoAcceptedDocumentType model.Error = "no accepted document type"
+
+// LinkGeminiWallet links a wallet to a Gemini account.
 func (service *Service) LinkGeminiWallet(ctx context.Context, walletID uuid.UUID, verificationToken, depositID string) (string, error) {
+	cl, err := service.Datastore.GetCustodianLinkByWalletID(ctx, walletID)
+	if err != nil && !errors.Is(err, model.ErrNoWalletCustodian) {
+		return "", handlers.WrapError(err, "failed to check linking mismatch", http.StatusInternalServerError)
```

Comment:
Question for understanding: What happens when `err` is `model.ErrNoWalletCustodian`? Just to make sure that the code below will work if that's the case.

Reply:
When the wallet does not have a wallet custodian i.e. `model.ErrNoWalletCustodian` it is not linked so it is either a new linking or it is in an unlinked state. It will basically returns `nil` (unlinked state), `nil` checks should be covered by `cl.isLinked()` checks further in the code. If it is unlinked then it is not an auth flow and it is a re-linking or new linking which are treated the same.

That method is an existing one and not very clear as part of the wallet refactor this will be improved so it is more obvious whats going on but it seemed like too big a change for this ticket.

Reply:
Thank for explaining, I recall now discussing this with you before. Yes, it makes sense, and yes, it would be too much to make those changes as part of this PR, so no worries.